linode/vpcips/framework_datasource.go

```diff
@@ -0,0 +1,113 @@
+package vpcips
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/v2/linode/helper"
+)
+
+func NewDataSource() datasource.DataSource {
+	return &DataSource{
+		BaseDataSource: helper.NewBaseDataSource(
+			helper.BaseDataSourceConfig{
+				Name:   "linode_vpc_ips",
+				Schema: &frameworkDataSourceSchema,
+			},
+		),
+	}
+}
+
+type DataSource struct {
+	helper.BaseDataSource
+}
+
+func (r *DataSource) Read(
+	ctx context.Context,
+	req datasource.ReadRequest,
+	resp *datasource.ReadResponse,
+) {
+	tflog.Debug(ctx, "Read data.linode_vpc_ips")
+
+	var data VPCIPFilterModel
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	id, d := filterConfig.GenerateID(data.Filters)
+	if d != nil {
+		resp.Diagnostics.Append(d)
+		return
+	}
+	data.ID = id
+
+	var result []any
+
+	// Check if VPC ID is provided
+	if !data.VPCID.IsNull() {
+		tflog.Debug(ctx, "Filtering IPs for specific VPC", map[string]interface{}{
+			"vpc_id": data.VPCID.ValueInt64(),
+		})
+
+		result, d = filterConfig.GetAndFilter(
+			ctx, r.Meta.Client, data.Filters,
+			func(ctx context.Context, client *linodego.Client, filter string) ([]any, error) {
+				return ListVPCIPs(ctx, client, filter, int(data.VPCID.ValueInt64()))
```

Comment:
int64 to int conversion may required `helper.FrameworkSafeInt64ToInt`

Reply:
Good to know, thanks for the tip!